extensions/csv/src/main/java/io/deephaven/csv/densestorage/QueueWriter.java

```diff
@@ -0,0 +1,203 @@
+package io.deephaven.csv.densestorage;
+
+import io.deephaven.csv.containers.CharSlice;
+
+import java.util.function.BiFunction;
+import java.util.function.IntFunction;
+
+public class QueueWriter<TARRAY, TREADER> {
+    private final Object sync;
+    private QueueNode<TARRAY> tail;
+    protected final int blobSize;
+    private final IntFunction<TARRAY> arrayFactory;
+    private final BiFunction<Object, QueueNode<TARRAY>, TREADER> readerFactory;
+    private boolean allowReaderCreation;
+    private TARRAY genericBlock;
+    protected int begin;
+    protected int current;
+    protected int end;
+
+    protected QueueWriter(int blobSize,
+                          IntFunction<TARRAY> arrayFactory,
+                          BiFunction<Object, QueueNode<TARRAY>, TREADER> readerFactory) {
+        this.sync = new Object();
```

Comment:
Do we actually need a separate synchronization object?  We could set it to this I think (I am not advocating for you to explicitly switch to "this" synchronization method level, just asking if we could avoid the additional object and keep the sync reference just make it point to this)

Reply:
I have two rationales for this and they are both stylistic in nature. After you read my rationale, you can decide whatever you think is most appropriate and I will comply.

1. I have always felt that Java objects were a little "promiscuous" by allowing you to take a lock on anything. I always felt is was better style to devote a specific object to being a lock object. That is, being more strict even though the language allows you to be loosey-goosey.
2. If you're a super-purist, in a context that matters (i.e. not this one), taking a lock "this" can be a security problem because if some bad actor has your object, they can also take a lock on "this" and make your code deadlock.

Reply:
I think is fine to keep as is.  In a context where this is in some critical path of either performance or number of objects created I would push more, but not now.